sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultService.java

```diff
@@ -75,9 +85,43 @@ public DefaultService(ServiceSpec serviceSpec, Collection<Plan> plans) {
         register(serviceSpec, this.plans);
     }
 
+    private void lock() {
+        String rootPath = CuratorUtils.toServiceRootPath(serviceSpec.getName());
+        String lockPath = CuratorUtils.join(rootPath, "LOCK_PATH");
+        curatorMutex = new InterProcessMutex(curatorClient, lockPath);
+
+        LOGGER.info("Acquiring ZK lock...");
+        try {
+            if (!curatorMutex.acquire(10, TimeUnit.SECONDS)) {
+                LOGGER.error("Failed to acquire ZK lock.  Exiting.");
+                System.exit(1);
```

Comment:
May as well use the hardExit() wrapper call and add a SchedulerErrorCode value for it.

Optional: Replace SchedulerErrorCode enum with a class that has explicit integer values, as enum values as ints can be brittle. Something like:

```
public class SchedulerExitCode {

  [... list of public static final int's for the error code values? or an ImmutableMap? ...]

  public SchedulerExitCode(int value) {
    this.value = value;
  }

  public void exit() {
    LOGGER.error("Exiting with code {}", value);
    System.exit(value);
  }
}
```